src/hooks/useUsersList.js

```diff
@@ -1,12 +1,25 @@
 import { useEffect, useState } from "react";
-import { getUsers } from "../service/getUsers";
+import { useDispatch, useSelector } from "react-redux";
+import { getUsersList } from "../service/getUsers";
+import { setUsersList } from "../redux/reducer/userReducer";
+import { selectUserList } from "../redux/selectors";
 
 export default function useUsersList() {
+  const dispatch = useDispatch();
+  const userList = useSelector(selectUserList);
   const [list, setList] = useState([]);
+
+  useEffect(() => {
+    setList(userList);
+  }, [userList]);
+
   useEffect(() => {
-    getUsers()
-      .then((res) => setList(res.data))
-      .catch(() => setList([]));
-  }, []);
+    getUsersList()
+      .then((res) => {
+        dispatch(setUsersList(res.data));
```

Comment:
why do you need it in the redux store? you set it into the component state, so you have two data sources